azure-pipelines.yml

```diff
@@ -57,6 +57,114 @@ stages:
       value: $(Build.SourceBranchName)
 
   jobs:
+  - job: get_impacted_area
+    displayName: "Get impacted area"
+    timeoutInMinutes: 240
+    continueOnError: true
+    pool: sonic-ubuntu-1c
+    steps:
+      - script: |
+          set -x
+
+          DIFF_FOLDERS=$(git diff HEAD^ HEAD --name-only | xargs -n1 dirname | sort -u | tr '\n' ' ')
+          echo -n "##vso[task.setvariable variable=DIFF_FOLDERS]$DIFF_FOLDERS"
+        displayName: "Get diff folders"
+
+      - script: |
+          set -x
+
+          pip install PyYAML
+          pip install natsort
+
+          sudo apt-get update && sudo apt-get install -y jq
```

Comment:
If the apt-get commond was running background(vm init), there may be a confict but low chance to happen.  Add a timeout maybe better.
    # The -o DPkg::Lock::Timeout=300 option sets the timeout for acquiring the package lock to 300 seconds.
    sudo apt-get -o DPkg::Lock::Timeout=300 upgrade
    sudo apt-get -o DPkg::Lock::Timeout=300 install -y jq